### DISCUSSION TITLE
Few issues / Clarifications on Elsa 2.0

### DISCUSSION THREAD

**Discussion by VivekKachare007** [elsa-workflows/elsa-core#1409] [Q&A]
1) Is there any hardware software specification for Elsa 2.0. What I observed with Elsa 1.0 as the data increases exponentially there is performance issues observed

2) Can we have cascading dropdown list in custom activities ?

**Comment by sfmskywalker** [comment]
1. This varies by application, the same way it varies what you need for a given ASP.NET application. Some require less, others require more. Elsa doesn’t need much, if anything, in addition. The data increase you mention still exists in Elsa 2, and is caused by storing activity execution log records in a table. These records can be deleted easily if disk space is a concern. We also have an open issue to enable retention policies for automatic cleanup.
2. Yes, you can implement custom controls and associate them with custom UIHints. This process is not documented, but an example can be found in index.html of the designer project.

**Comment by VahidShir** [comment]
@sfmskywalker I couldn't find that index.html file. Can you give a link of that?

**Comment by sfmskywalker** [comment]
Sure, here you go: https://github.com/elsa-workflows/elsa-core/blob/master/src/designer/elsa-workflows-studio/src/index.html

**Comment by sfmskywalker** [comment]
@VivekKachare007 In case you're looking for an example of implementing cascading dropdowns, I posted [this SO answer](https://stackoverflow.com/questions/72102024/elsa-workflow-how-to-implement-cascade-dropdown/72273275#72273275) that might help.
